src/Moderation/Entity/Report.php

```diff
@@ -35,24 +35,30 @@ class Report extends \Friendica\BaseEntity
 	/** @var int|null */
 	protected $id;
 	/** @var int ID of the user making a moderation report*/
+	protected $reporterId;
+	/** @var int ID of the contact making a moderation report*/
 	protected $uid;
 	/** @var int ID of the contact being reported*/
 	protected $cid;
 	/** @var string Optional comment */
 	protected $comment;
+	/** @var string Optional category */
+	protected $category;
 	/** @var bool Whether this report should be forwarded to the remote server */
 	protected $forward;
 	/** @var \DateTime|null When the report was created */
 	protected $created;
 	/** @var array Optional list of URI IDs of posts supporting the report*/
 	protected $postUriIds;
 
-	public function __construct(int $uid, int $cid, \DateTime $created, string $comment = '', bool $forward = false, array $postUriIds = [], int $id = null)
+	public function __construct(int $uid = null, int $reporterId, int $cid, \DateTime $created, string $comment = '', string $category = null, bool $forward = false, array $postUriIds = [], int $id = null)
```

Comment:
If you intent on keeping the `uid` field around, please move the optional parameter after the mandatory ones.
```suggestion
	public function __construct(int $reporterId, int $cid, \DateTime $created, int $uid = null, string $comment = '', string $category = null, bool $forward = false, array $postUriIds = [], int $id = null)
```
Please be sure to update the places the constructor is called.

Reply:
done